aws/sns/resources.py

```diff
@@ -21,20 +22,49 @@ def sns_topics():
     )
 
 
-def sns_subscription_attributes():
+def sns_parent_topic_exists(subscription):
     "https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/sns.html#subscription"
-    return [
+    topicArn = ":".join(
+        re.split(":", subscription)[0:6]
+    )  # Naive method of divining topic from subscription
+    try:
         botocore_client.get(
             service_name="sns",
-            method_name="get_subscription_attributes",
+            method_name="get_topic_attributes",
             call_args=[],
-            call_kwargs={"SubscriptionArn": subscription["SubscriptionArn"]},
-            profiles=[subscription["__pytest_meta"]["profile"]],
-            regions=[subscription["__pytest_meta"]["region"]],
+            call_kwargs={"TopicArn": topicArn},
+        )
+        return True
+    except:
+        return False
+
+
+def sns_get_subscription_attrs(subscription):
+    "https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/sns.html#subscription"
+    # This function catches the error thrown by the AWS API when attempting to query the attributes of
+    # a subscription that has no parent topic.
+    try:
+        attrs = (
+            botocore_client.get(
+                service_name="sns",
+                method_name="get_subscription_attributes",
+                call_args=[],
+                call_kwargs={"SubscriptionArn": subscription["SubscriptionArn"]},
+                profiles=[subscription["__pytest_meta"]["profile"]],
+                regions=[subscription["__pytest_meta"]["region"]],
+            )
+            .extract_key("Attributes")
+            .values()[0]
         )
-        .extract_key("Attributes")
-        .values()[0]
-        for subscription in sns_subscriptions()
+        return attrs
+    except:
+        pass
+
+
+def sns_subscription_attributes():
+    "https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/sns.html#subscription"
+    return [
+        sns_get_subscription_attrs(subscription) for subscription in sns_subscriptions()
```

Comment:
The function `sns_get_subscription_attrs()` defined above will return `None` whenever it encounters and ignores an error, so there would be random `None` values in this list, which I don't think is the intended behaviour.

Reply:
I'm not certain what the best thing to return here would be.  It's unable to check the PendingVerification status, so maybe we still want these to fail.

Reply:
It depends on what you want. Currently, `sns_get_subscription_attrs()` will return `None` on error, which may be reasonable behaviour. If it is, you probably want to filter the None values out from the list before returnning it, e.g. with `[x for x in my_list if x is not None]`.

Or maybe it's an actual error when you can't retrieve details for one of the subscriptions? In that case, don't catch it, or at least don't catch it in `sns_get_subscription()`, but further up the call stack, where you know what to do with it.

Or maybe you can still get details for these subscriptions by asking the API in a different way? I just don't know, since I don't understand what that error condition ("unable to check the PendingVerification status") means.

Reply:
From my understanding, we'd probably want to go with the `[x for x in my_list if x is not None]` method.